habitat_sim/agent/agent.py

```diff
@@ -54,18 +59,20 @@ class SixDOFPose(object):
     :property rotation: unit quaternion rotation
     """
 
-    position: np.ndarray = np.zeros(3)
+    position: np.ndarray = attr.ib(factory=_triple_zero, validator=all_is_finite)
     rotation: Union[np.quaternion, List] = np.quaternion(1, 0, 0, 0)
 
 
 @attr.s(auto_attribs=True, slots=True)
 class AgentState(object):
-    position: np.ndarray = np.zeros(3)
+    position: np.ndarray = attr.ib(factory=_triple_zero, validator=all_is_finite)
     rotation: Union[np.quaternion, List, np.ndarray] = np.quaternion(1, 0, 0, 0)
```

Comment:
Let's also check for the rotation being unit length. `.norm()` for `np. np.quaternion` and `np.linalg.norm` for the others.

Reply:
Are they any np.quaternion functions to do this @erikwijmans ? A lot of numpy functions like np.finite don't work on that type.

Reply:
`.norm()` will do it.  That will not be 1 when there is inf or nan